jdbc-dbcp/src/main/java/io/micronaut/configuration/jdbc/dbcp/DatasourceConfiguration.java

```diff
@@ -45,7 +47,8 @@
  */
 @Context
 @EachProperty(value = BasicJdbcConfiguration.PREFIX, primary = "default")
-public class DatasourceConfiguration extends BasicDataSource implements BasicJdbcConfiguration {
+@Internal
+class DatasourceConfiguration extends BasicDataSource implements BasicJdbcConfiguration {
```

Comment:
this is a breaking change

Reply:
I see, reverted and changed `setEnabled` method non public with docs about the behavior and intention of the method.